### PR TITLE
improve: skip server selection and update docs for auto-login

### LINKED ISSUE #2758: Don't prompt in `infra login` when info is already known
This is related to #2557 but I think it's a distinct issue.

I would like for subsequent runs of `infra login` after the first setup to do the needful without prompting me.

For some background, I initially configured infra by running `infra login --provider okta infra.example.com`.  I only use the one server and the one login method, and I find it annoying that each subsequent time I need to run `infra login` that I either (a) need to provide the server and provider again or (b) hit enter through some prompts.

```
$ infra login
? Select a server:  [Use arrows to move, type to filter]
> infra.example.com
  Connect to a new server

? Select a login method:  [Use arrows to move, type to filter]
> okta (example.okta.com)
  Login with username and password
```

My preferred solution would be for `infra login` to use the saved config and do it without prompting if there is a previously configured server.  An also acceptable solution would be to have a flag to do this, like `infra login -y`.  My argument for the former is that the vast majority of the time people will only have a single login and want to use it (or log into their most recently used server if there's more than one).  Also this would match other tooling that performs similar things like `kubectl oulogin`.

## Environment Details

```
$ infra version
 Client: 0.14.2
 Server: 0.14.1
```

```
$ kubectl version
Client Version: v1.24.3
Kustomize Version: v4.5.4
Server Version: v1.22.8
WARNING: version difference between client (1.24) and server (1.22) exceeds the supported minor version skew of +/-1
```

### PR DESCRIPTION
**Pull request by j-sneh**
## Summary

Users have requested being able to run `infra login` and get logged in without the survey prompts. One could already set the `INFRA_SERVER` env var to skip server selection. This PR adds the `INFRA_PROVIDER` and cli docs to make it more clear. It also uses local login if the provider is `"infra"`

## Checklist

- [ ] Wrote appropriate unit tests
- [ ] Considered security implications of the change
- [x] Updated associated docs where necessary
- [ ] Updated associated configuration where necessary
- [x] Change is backwards compatible if it needs to be (user can upgrade without manual steps?)
- [x] Nothing sensitive logged
- [x] Considered data migrations for smooth upgrades

## Related Issues

Resolves #2758

### CODE REVIEW

File: internal/cmd/login.go [hunk 1]
```diff
@@ -78,6 +83,10 @@ $ infra login`,
 				options.Server = args[0]
 			}
 
+			if options.Provider == "infra" {
```
Comment (BruceMacD) [inline internal/cmd/login.go:86]: The infra provider doesn't seem to work for me. Me guess is setting this blank doesnt see that the provider is set

Comment (j-sneh) [reply internal/cmd/login.go:86]: This is the local login. The idea here is if someone has an `INFRA_PROVIDER` defined, if they want to log in to a local user then they specify `--provider infra`. Maybe `"infra"` shouldn't be a valid provider name?

Comment (BruceMacD) [reply internal/cmd/login.go:86]: Infra as a provider name for this functionality seems good to me. Im just having issues with getting this behavior to work, I may be doing something wrong.

It still prompts me for the provider, but I expect it should just automatically prompt me for username/password:
```
$ export INFRA_PROVIDER=infra 

$ infra login localhost                 
? Select a login method:  [Use arrows to move, type to filter]
> google (accounts.google.com)
  okta (dev.okta.com)
  Login with username and password

 $ infra login localhost --provider infra
? Select a login method:  [Use arrows to move, type to filter]
> google (accounts.google.com)
  okta (dev-50966300.okta.com)
  Login with username and password
```

Comment (j-sneh) [reply internal/cmd/login.go]: @BruceMacD That worked locally because my local infra instance only had local login. This should be fixed now.

**Review by BruceMacD — APPROVED** [review]

**Pull request merged**